syncs/shardedint_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package syncs
+
+import (
+	"expvar"
+	"sync"
+	"testing"
+)
+
+func BenchmarkShardedInt(b *testing.B) {
+	b.ReportAllocs()
```

Comment:
you have this, but also lock in no allocs with a test using testing.AllocsPerRun and verify it's zero